folding/validators/hyperparameters.py

```diff
@@ -4,34 +4,43 @@
 
 import bittensor as bt
 
+from folding.utils.openmm_forcefields import FORCEFIELD_REGISTERY
+
 
 class HyperParameters:
     def __init__(self, exclude: Union[Dict[str, str], List[str]] = None) -> None:
         """Sample hyperparameters for protein folding for a specific pdb_id
 
         Args:
             exclude (List[str], optional): List of hyperparameters to exclude. Defaults to None.
-                needs to be either ['FF','BOX','WATER'] to exclude a specific hyperparameter, or
+                needs to be either ['FF','WATER'] to exclude a specific hyperparameter, or
                 you can exclude a specific value(s) by passing {'FF': 'charmm27', 'BOX_TYPE': 'dodecahedron'}.
         """
 
-        # Need to download files for specific inputs.
-        self.FF: List[str] = ["charmm27", "amber03"]
-        self.BOX: List[str] = ["dodecahedron", "octahedron", "cubic"]
-        self.WATER: List = ["tip3p", "spce"]
-        self.BOX_DISTANCE: List[float] = [1.0]
-
-        self.parameter_set = {
-            "FF": self.FF,
-            "BOX": self.BOX,
-            "WATER": self.WATER,
-            "BOX_DISTANCE": self.BOX_DISTANCE,
-        }
+        self.sampled_combinations: List[Dict[str, str]] = []
+        self.all_combinations: List[Dict[str, str]] = []
 
+        # Need to download files for specific inputs.
+        fields = [field() for field in FORCEFIELD_REGISTERY.values()]
         self.exclude: List[str] = exclude or []
 
-        self.create_parameter_space()
-        self.setup_combinations()
+        for field in fields:
+            self.FF = field.forcefields()
+            self.WATER = field.waters()
+            self.BOX_DISTANCE = [1.0]
+
+            parameter_set = {
+                "FF": self.FF,
+                "WATER": self.WATER,
+                "BOX_DISTANCE": self.BOX_DISTANCE,
+            }
+
+            self.create_parameter_space()
+            self.all_combinations.extend(
+                self.setup_combinations(parameter_set=parameter_set)
+            )
+
+        self.TOTAL_COMBINATIONS = len(self.all_combinations)
```

Comment:
Make sure to sample recommended combinations first before other combinations

Reply:
DONE